chore(docs): include the environment variables section in the utilities documentation

barreeeiroo opened the pull request:
<!-- markdownlint-disable MD041 MD043 -->
**Issue number:** https://github.com/aws-powertools/powertools-lambda-python/issues/2923
**Supersedes:** https://github.com/aws-powertools/powertools-lambda-python/pull/2922

## Summary

Clarifies some documentation pages by including their related environment variables. They were only included on the main page, though it may be useful to also mention them on their corresponding page.

### Changes

For each page, a new _Environment variables_ section was created under _Getting started_ right before _Advanced_. This section contains a table listing all environment variables affecting that page. It also mentions the preference order (parameter overrides environment variable).

### User experience

When navigating through documentation, developers will now find the environment variables for each core/utility on its page. It is useful as it allows us to quickly understand the behaviour of these environment variables by <kbd>Ctrl</kbd>+<kbd>F</kbd>'ing them.

This comes from personal experience, as I found out about Tracer's environment variables by checking the parameters of Tracer decorators' source code. But when going to Tracer's documentation page, I couldn't find these environment variables, until I realized they were only in the main page.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

No.

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

barreeeiroo committed:
Include Tracer environment variables

Clarifies Tracer's documentation page by including its related environment variables.

barreeeiroo committed:
Include Logger environment variables

Clarifies Logger's documentation page by including its related environment variables.

barreeeiroo committed:
Include Metrics environment variables

Clarifies Metrics' documentation page by including its related environment variables.

barreeeiroo committed:
Include middleware factory environment variables

Clarifies middleware factory's documentation page by including its related environment variables.

barreeeiroo committed:
Include parameters utility environment variables

Clarifies parameters utility's documentation page by including its related environment variables.

leandrodamascena committed:
Merge branch 'develop' into docs/env-vars-section

leandrodamascena commented:
Thanks for sending this PR @barreeeiroo!!

Hei @heitorlessa! Since I don't have the context here, I'd like your review or we can sync tomorrow and I can review.

leandrodamascena commented on docs/utilities/middleware_factory.md:
```diff
@@ -19,6 +19,16 @@ Middleware factory provides a decorator factory to create your own middleware to
 
 You might need a custom middleware to abstract non-functional code. These are often custom authorization or any reusable logic you might need to run before/after a Lambda function invocation.
 
+### Environment variables
```
What do you think of the "Environment Variables" section being the last of the Getting Started section? So we keep a standard.

barreeeiroo replied:
Yeah, my bad, I was actually placing all of them at the end of _Getting Started_ right before _Advanced_, but I probably missed that one.

leandrodamascena replied:
Thanks for addressing this feedback @barreeeiroo! :rocket:

leandrodamascena commented on docs/core/logger.md:
````diff
@@ -341,6 +341,18 @@ You can easily change the date format using one of the following parameters:
     --8<-- "examples/logger/src/date_formatting_output.json"
     ```
 
+### Environment variables
````
In logger.md we already have a section that explains some ENV variables (https://docs.powertools.aws.dev/lambda/python/2.22.0/core/logger/#getting-started). How about adding a disclaimer like: "check for additional environment variables here" and adding a link to this new section?

barreeeiroo replied:
Yep, that makes sense. Will add it 👍
I've also seen that Metrics also mentions a few environment variables for the quick start, though it is only 1 environment variable so there are no other variables in the corresponding section.

leandrodamascena reviewed (changes requested):
Hello @barreeeiroo! I made a couple of small comments and after that I think we're ready to merge!

Thank you very much.

barreeeiroo committed:
Move environment variables section

It should be the last one under Getting Started and right before Advanced.

barreeeiroo committed:
Link basic configuration to environment variables

As basic configuration mentions a few environment variables, it may be useful to provide a link to all the other available ones.

leandrodamascena committed:
Merge branch 'develop' into docs/env-vars-section

leandrodamascena reviewed (approved):
Fantastic work @barreeeiroo! This will make it easier for our customers to find environment variables that can be used in each utility.

![image](https://github.com/aws-powertools/powertools-lambda-python/assets/4295173/e7f2b4da-de80-47e2-ae52-41ed398d8203)
